framework/scripts/wazuh_server.py

```diff
@@ -378,19 +382,105 @@ def start():
         main_function = worker_main
 
     try:
-        asyncio.run(main_function(args, server_config, main_logger))
+        loop = asyncio.new_event_loop()
+        loop.add_signal_handler(signal.SIGTERM, partial(stop_loop, loop))
+        loop.create_task(monitor_server_daemons(loop, psutil.Process(server_pid)))
+        loop.run_until_complete(main_function(args, server_config, main_logger))
     except KeyboardInterrupt:
         main_logger.info('SIGINT received. Shutting down...')
     except MemoryError:
         main_logger.error("Directory '/tmp' needs read, write & execution " "permission for 'wazuh' user")
     except WazuhDaemonError as e:
         main_logger.error(e)
+    except RuntimeError:
+        main_logger.info('Main loop stopped.')
     except Exception as e:
         main_logger.error(f'Unhandled exception: {e}')
     finally:
         shutdown_server(server_pid)
 
 
+def check_daemon(proc_list: list, proc_name: str, children_number: int):
+    """Check the daemon is in the list of process and have the correct number of children.
+
+    Parameters
+    ----------
+    proc_list : list
+        List of children process to search within.
+    proc_name : str
+        Name of the process to check.
+    children_number : int
+        Expected number of children to check.
+
+    Raises
+    ------
+    WazuhDaemonError
+        When the process does not have the correct number of children process.
+    """
+    child: psutil.Process = [i for i in proc_list if proc_name[:-1] in i.name()][0]
+    if child.status() == psutil.STATUS_ZOMBIE:
+        main_logger.error(f"Daemon `{proc_name}` is not running, stopping the whole server.")
+        clean_pid_files(proc_name)
+        return
+    if len(child.children()) != children_number:
+        raise WazuhDaemonError(f'Daemon `{proc_name}` does not have the correct number of children process.')
+
+
+async def monitor_server_daemons(loop: asyncio.BaseEventLoop, server_process: psutil.Process):
+    """Monitor the status of the server daemons.
+
+    Parameters
+    ----------
+    loop : asyncio.BaseEventLoop
+        The loop to stop in case any of the daemons does not meet the expected status.
+    server_pid : int
+        PID of the server to get the children.
+    """
+    comms_api_config = CentralizedConfig.get_comms_api_config()
+    proc_tree = {
+        MANAGEMENT_API_DAEMON_NAME: {"children": 3},
+        COMMS_API_DAEMON_NAME: {"children": comms_api_config.workers + 4},
+        ENGINE_DAEMON_NAME: {"children": 0}
+    }
+
+    while True:
+        await asyncio.sleep(10)
+        child_proceses = server_process.children()
+
+        try:
+            for key, value in proc_tree.items():
+                check_daemon(child_proceses, key, value['children'])
```

Comment:
Following the previous comments

```suggestion
        try:
            for proc_name, children in proc_tree.items():
                check_daemon(child_processes, proc_name, children)
```